decoder.go

```diff
@@ -6,6 +6,20 @@ import (
 	"io"
 )
 
+const (
+	// MaximumRecordLength defines the maximum length in bytes, acceptable for records while decoding.
+	// This maximum prevents from excessive memory allocation for decoding.
+	// The value is derived from MAX_PKT_SIZ 65536 in sflow reference implementation
+	// https://github.com/sflow/sflowtool/blob/bd3df6e11bdf8261a42734c619abfe8b46e1202f/src/sflowtool.c#L4313
+	MaximumRecordLength = 65536
+
+	// MaximumHeaderLength defines the maximum length in bytes, acceptable for packet flow samples while decoding.
+	// This maximum prevents from excessive memory allocation for decoding.
+	// The value is derived from INM_MAX_HEADER_SIZE 256 in sflow reference implementation
+	// https://github.com/sflow/sflowtool/blob/bd3df6e11bdf8261a42734c619abfe8b46e1202f/src/sflowtool.h#L28
+	MaximumHeaderLength = 256
```

Comment:
I could not find any information about the maximum in the sflow specification. First my setting was 1500 and I corrected it in the second commit based on the sflowtool source code. I am fine with 1300 or 1500.  
The main issue is, the header length field in the sflow package is uint32, so the maximum would be 0xFFFFFFFF, respectively 4,294,967,295 bytes. And this is by far to much for the `make([]byte, f.HeaderSize+padding)`.